### ISSUE TITLE
Improve performance of ExtractFrame method 

### ISSUE THREAD

**Issue by kahnivore** [RenderHeads/UnityPlugin-AVProVideo#605]
Calling ExtractFrameAsync causing poor performance with large video sizes or list of videos. 
Request to Improve performance playback on extracting a frame.

**Comment by AndrewRH** [comment]
Yes this currently blocks the main thread while grabbing the texture...  We will switch to Unity's [AsyncGPUReadbackRequest ](https://docs.unity3d.com/ScriptReference/Rendering.AsyncGPUReadbackRequest.html)

We also plan to move all the logic for frame grabbing into it's own class, for modularity.

There should also be a demo scene for this.
